### DISCUSSION TITLE
Sprint Qualifying - driver.grid, driver.position

### DISCUSSION THREAD

**Discussion by marinik98** [theOehrly/Fast-F1#99] [Ideas]
As we can have access to the starting grid position of a driver for a race and the final position of that driver, can we also have access to the same attributes but for the Sprint Qualifying?

**Comment by siddharth-tewari** [comment]
@marinik98 You can use the qualifying data for race and fp2 for sprint but I don't know how to check for grid penalties.

**Reply by siddharth-tewari** [reply]
Update: I found out that you can use <a href="http://ergast.com/api/f1/2021/22/results">http://ergast.com/api/f1/2021/22/results</a>
I will try to make a program and share it here. I don't about how to do it for Sprint Qualifying.

**Reply by marinik98** [reply]
I mean, I get I can get the list of starters from the result of the quali, but as you said, this is not the case, as grid penalties are a big factor, maybe I’ll just slap a csv for the three races and that’s it. The problem should not be posed for sprint qualifying results, as the .grid gives the .position of the sprint qualifying, right?

**Reply by siddharth-tewari** [reply]
@marinik98 I think FP2 results give for Sprint Qualifying and you can take starting grid from the ergast links.

**Comment by theOehrly** [comment]
I will try to add this for next season. I need to improve support for sprint qualifying in general.

**Comment by theOehrly** [comment]
This is now implemented in the second pre-release version of v2.2 (stable will follow later this week). The new `SessionResults` object can be accessed through `Session.results` after calling `Session.load`. 
`Session.get_driver(...)` will now return a `DriverResults` object. 
Postion, grid position, qualifying times and total race time will be given if applicable for the respective session.
